test/Npgsql.Tests/CommandBuilderTests.cs

```diff
@@ -50,8 +53,17 @@ public void DeriveParametersVarious()
                 NpgsqlCommandBuilder.DeriveParameters(cmd);
                 Assert.That(cmd.Parameters, Has.Count.EqualTo(3));
                 Assert.That(cmd.Parameters[0].Direction, Is.EqualTo(ParameterDirection.Input));
+                Assert.That(cmd.Parameters[0].NpgsqlDbType, Is.EqualTo(NpgsqlDbType.Integer));
```

Comment:
Probably rename this test to `DeriveFunctionParametersVarious`?

Reply:
Agreed.
I took the opportunity to homogenize all the test names and added one more test that documents the currently non-working behavior of parameters that are arrays of composite types.